io-engine/src/grpc/v1/stats.rs

```diff
@@ -30,20 +30,14 @@ pub struct StatsService {
 }
 
 #[async_trait::async_trait]
-impl<F, T> SvcLocker<F, T> for StatsService
+impl<F, T> Serializer<F, T> for StatsService
 where
     T: Send + 'static,
     F: core::future::Future<Output = Result<T, Status>> + Send + 'static,
 {
-    async fn svc_lock(
-        &self,
-        ctx: GrpcClientContext,
-        pool_svc: PoolService,
-        replica_svc: ReplicaService,
-        f: F,
-    ) -> Result<T, Status> {
-        let pool_context = pool_svc.client_context.write().await;
-        let replica_context = replica_svc.client_context.write().await;
+    async fn locked(&self, ctx: GrpcClientContext, f: F) -> Result<T, Status> {
+        let pool_context = self.pool_svc.client_context.write().await;
```

Comment:
read locks should be sufficient here isn't it?

Reply:
We do Stats reset here which is a Mutating Operation on Bdev. Hence decided to take write lock to be safe.

Im thinking Reset would not impact Bdev and other part of Pool/Replica. Do you think we could allow Read operation safely while this is going on?